### ISSUE TITLE
Requirements for a feature staying in the repo

### ISSUE THREAD

**Issue by gabrieldemarmiesse** [keras-team/keras-contrib#418]
This is to track the work necessary to do on features already there.

A feature should:
* Have a proper docstring
* Have a link to the paper it comes from in the docstring.
* Be displayed in the documentation
* Have an API which correspond to the one used in keras (eg. use `lr` for the learning rate in optimizers)
* Be in a single file
* Have unit tests
* Those unit tests should be in a single file too.
* Those unit tests should run in the CI system (here travis).
* Should NOT be added automatically to the custom object of keras.
* Only use the keras public API (otherwise it won't work with tf.keras)
* Have a codeowner for the feature and the tests in the `CODEOWNERS` file.

Features which do not respect this will be removed / not included in this repo. 

Otherwise, we'll end up like `tensorflow.contrib` :(

**Comment by titu1994** [comment]
Ok that makes sense.

**Comment by gabrieldemarmiesse** [comment]
I hope we can make a cool v1.0 :)

**Comment by titu1994** [comment]
Yes, its been a long time coming for this contrib repo. :D
